code_annotations/cli.py

```diff
@@ -109,7 +112,36 @@ def static_find_annotations(config_file, source_path, report_path, verbosity):
         None
     """
     try:
+        start_time = datetime.datetime.now()
         searcher = StaticSearch(config_file, source_path, report_path, verbosity)
-        searcher.search()
+        all_results = searcher.search()
+
+        if lint:
+            click.echo("Performing linting checks...")
+            # Check grouping and choices
+            searcher.check_results(all_results)
+
+            # If there are any errors, do not generate the report
+            if searcher.errors:
+                click.secho("\nSearch failed due to linting errors!", fg="red")
+                click.secho("{} errors:".format(len(searcher.errors)), fg="red")
+                click.secho("---------------------------------", fg="red")
+                click.echo("\n".join(searcher.errors))
+                exit(-1)
+            click.echo("Linting passed without errors.")
+
+        if report:
+            click.echo("Writing report...")
```

Comment:
Is there anything useful to report if the linting isn't performed?

Reply:
The reporting here is actually writing out the yaml file, so if someone doesn't really care about enforcing the rules w/groups and choices they could just run the reports without the checks.